### PR TITLE
Use the Go Jsonnet port instead of the C++ port

### PR DESCRIPTION
**Pull request by seh**
The Jsonnet project's development focus has shifted to [the Go language port](https://github.com/google/go-jsonnet), with all the recent performance optimizations landing there. Change _rules_jsonnet_ to use the Go port instead of the C++ port unconditionally.

While it would be possible to expose this choice to users of this package, perhaps by way of introducing a sibling function to `jsonnet_repositories`, or introducing a parameter to that function, I don't think it's worth the maintenance burden to do so.

@sparkprime and @sbarzowski, your guidance would be welcome here.

### CODE REVIEW

**Comment by seh** [comment]
It looks like the transitive dependencies for _google/go-jsonnet_ (in particular, on _rules_go_) aren't picked up properly here. I'll look into that.

**Comment by Globegitter** [comment]
In my old job we already switched over to the go version because we hat compiling issues on some non LTS Ubuntu versions. While I am generally in favour of this change I also think we should be a bit more careful with it, e.g. now people need to pull in the go toolchain which they might have not have needed before. Also now the c++ version of the formatter is gone and we should think if we should pull that in separately.

So even though we are not on 1.0 yet I would still consider it a breaking change and maybe even posting something on bazel-dev or bazel-discuss. WDYT?

**Comment by seh** [comment]
I figured out what was keeping the tests from passing here. I opened google/go-jsonnet#292, which I'll show use of from here as well shortly.

**Comment by seh** [comment]
@Globegitter, where does this repository use _jsonnetfmt_?

As you said, I do consider this to be a breaking change, at least to consuming projects' _WORKSPACE_ files. The use of the `jsonnet_library`, `jsonnet_to_json`, and `jsonnet_to_json_test` rules shouldn't need to change at all.

The release numbers have been low here, seeming far away from 1.0. It's tempting to say that we could keep the existing _WORKSPACE_ "interface" to the C++ Jsonnet port the same, and introduce new macros to use the Go port instead, with the threat of deprecating the former. I don't know whether I'll have enough time in the coming week or two to work through that complexity, though.

File: jsonnet/jsonnet.bzl [hunk 1]
```diff
@@ -788,11 +788,10 @@ Example:
...
-    http_archive(
-        name = "jsonnet",
-        sha256 = "f6f0c4ea333f3423f1a7237a8a107c589354c38be8a2a438198f9f7c69b77596",
-        strip_prefix = "jsonnet-0.13.0",
-        urls = [
-            "https://github.com/google/jsonnet/archive/v0.13.0.tar.gz",
-        ],
+    git_repository(
+        name = "jsonnet_go",
+        # TODO: Update this once google/go-jsonnet#292 lands.
```
Comment (seh) [inline jsonnet/jsonnet.bzl]: NB: I'm publishing this here in advance. Some subsequent coordination is necessary.

File: README.md [hunk 2]
```diff
@@ -25,13 +25,22 @@ external repositories for Jsonnet:
 ```python
 http_archive(
     name = "io_bazel_rules_jsonnet",
+    # TODO: Update this to reflect a later release.
```
Comment (seh) [inline README.md]: NB: I'm publishing this here in advance. Some subsequent coordination is necessary.

**Comment by Globegitter** [comment]
Yeah I know that this repository has not been depending on the formatter, but at my previous company we where depending on it and other people might have made use of it as well, so it might still be nice to provide the c++ formatter for those cases.

Yeah I like the idea of keeping both at the same time for a migration window. I am happy to deprecate the c++ version immediately though. I won't have much time the next couple of weeks, but might also be able to spend some time on this after if you don't get around to it.

**Comment by seh** [comment]
@Globegitter, please see my most recent commit a0444b9c25c3fa213ff213f6a881f5a6eb38cee9. It makes both ports of Jsonnet available, with selection via command-line flag. I didn't see an easier way to use [Bazel's configurable build attributes system](https://docs.bazel.build/versions/master/configurable-attributes.html) without relying on a command-line flag.

The default choice is the Go port. If you feel strongly that we should retain the C++ port as the default choice, I can do that. My proposal here—including the deprecation warning in the _README.md_ file—indicates my preference for switching to the Go port.

**Comment by seh** [comment]
Note that we could add [more CI targets to the _.bazelci/presubmit.yml_ file](https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/README.md#using-specific-build--test-flags) to exercise this repository with the possible `--define jsonnet_port=(cpp|go)` flag values. It would be repetitious, doubling the size of that file, but could catch problems faster. Let me know if you'd like me to add those.

**Comment by seh** [comment]
I've been using this to good effect in a couple of projects of mine. Has anyone else tried it? Is there anything to adjust before merging it?

**Comment by Globegitter** [comment]
@seh Sorry this dropped of my radar I will take a look again in the next couple of days.

**Comment by Globegitter** [comment]
@seh This looks good to me can we just swap the default back to cpp for now, so we can release it fully backwards-compatible and then we can change it in a subsequent commit and make another release. So people have one release to test things out and report any potential issues.

**Comment by seh** [comment]
> @seh can we just swap the default back to cpp for now, so we can release it fully backwards-compatible

Yes, I think it will be a small code change to do so. It's slightly more work to adjust the documentation, but I'll take care of it. I will share a separate patch (along this same branch) for the change in default, so that we retain the current head as a destination for the follow-on release when we really commit to the Go port being the default.

**Comment by seh** [comment]
Please see commit 067095509a5d7a75ff013ea63f60300fb3975673 for the requested change in the default Jsonnet port.

**Comment by Globegitter** [comment]
@seh Sorry for the delay,and thanks for the great contribution.

**Comment by Globegitter** [comment]
@seh sorry again for the delay, I also released version 0.2.0 now. If you want you can already submit a PR to flip the default back again and if there are now issues reported I can release it in one or two weeks time.

**Comment by seh** [comment]
Thank you! I expect that I can post that follow-on patch to switch the default within a couple of days.

**Comment by seh** [comment]
See #129 for following through on changing the default Jsonnet port.

**Pull request merged**